### PR TITLE
improve readability of fw_gloabl_config

### PR DESCRIPTION
**Pull request by cdknorow**
- move user settings to global configs from sensor_audio_config and sensor_ssh
- update sml_output to not include features by default (larger models were causing an overflow)
- organize fw_global_config by things a user needs to change, add headers to improve readability
- create new S3_FIRMWARE_DATASAVE attribute which enables the SD card capture

### CODE REVIEW

**Comment by cdknorow** [comment]
@murthyvedula  this is ready for review now. Can you double check the datasave logic

File: qf_apps/qf_ssi_ai_app/src/ssi_comms.c [hunk 1]
```diff
@@ -98,7 +98,7 @@ void ssiTaskHandler(void* pParameter)
                sensor_ssss_startstop(1);
 #endif
 #if (SSI_SENSOR_SELECT_AUDIO)
-               uart_tx_raw_buf(UART_ID_SSI, "AUDIOSTREAMSTART", 16);
+               uart_tx_raw_buf(UART_ID_APP, "AUDIOSTREAMSTART", 16);
```
Comment (murthyvedula) [inline qf_apps/qf_ssi_ai_app/src/ssi_comms.c:101]: DCL will need to ignore these bytes indicating start of audio data stream

File: qf_apps/qf_ssi_ai_app/src/main.c [hunk 2]
```diff
@@ -73,13 +73,17 @@ uint8_t DeviceClassUUID[UUID_TOTAL_BYTES] =
 
 static const fw_global_config_t fw_global_config_vars =
 {
+#if (SSI_SENSOR_SELECT_AUDIO)
```
Comment (murthyvedula) [inline qf_apps/qf_ssi_ai_app/src/main.c:76]: no need for this #if wrapping

**Review by sivaQL — APPROVED** [review]

File: qf_apps/qf_ssi_ai_app/src/ssi_comms.c (hunk 1 above)
Comment (jublin) [reply qf_apps/qf_ssi_ai_app/src/ssi_comms.c:101]: Why are we writing these?

DCL will throw out the first read after sending connect, though

File: qf_apps/qf_ssi_ai_app/inc/app_config.h [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/*==========================================================
+ * Copyright 2020 QuickLogic Corporation
```
Comment (jublin) [inline qf_apps/qf_ssi_ai_app/inc/app_config.h:2]: This would need to be done for QT+ as well

File: qf_apps/qf_ssi_ai_app/src/ssi_comms.c (hunk 1 above)
Comment (murthyvedula) [reply qf_apps/qf_ssi_ai_app/src/ssi_comms.c:101]: the thought is this string "AUDIOSTREAMSTART" would serve to synchronize the audio data

**Pull request merged**